Sources/ArgumentParser/Completions/CompletionsGenerator.swift

```diff
@@ -41,6 +41,17 @@ public struct CompletionShell: RawRepresentable, Hashable, CaseIterable {
   public static var allCases: [CompletionShell] {
     [.zsh, .bash, .fish]
   }
+
+  /// An instance representing the shell for which completions are being
+  /// requested.
+  public internal(set) static var requesting: CompletionShell?
```

Comment:
Could you add a note to the documentation that this is only non-`nil` while generating shell scripts and custom completions?